radicle-node/src/service.rs

```diff
@@ -975,6 +983,48 @@ where
         Ok(false)
     }
 
+    /// A convenient method to check if we should fetch from a `RefsAnnouncement`
+    /// with `scope`.
+    fn should_fetch_refs_in_announcement(
+        &self,
+        message: &RefsAnnouncement,
+        scope: &tracking::Scope,
+    ) -> Result<bool, session::Error> {
+        let should_fetch = match scope {
```

Comment:
nit: I personally wouldn't bother with the `let` and just have the match arms result in an `Ok` or `Err`. That saves you from doing `return Err` below.

Reply:
In this particular case, I found `should_fetch` seems clearer to me. Again, this is only my personal preference. Either would work.

Reply:
One specific reason is that, this `match` handles a subset of errors differently, so I cannot use `?` to early return from calling `namespaces_for()`, otherwise it will better & simpler to return from each match arm.